ui/app/components/sidebar/frame.hbs

```diff
@@ -1,42 +1,44 @@
-<HcAppFrame::Frame @hasSidebar={{@showSidebar}} as |Frame|>
+<Hds::AppFrame @hasSidebar={{@showSidebar}} as |Frame|>
   <Frame.Sidebar data-test-sidebar-nav>
-    <HcAppFrame::Sidenav @contentNodeID="sidebar-nav" as |Sidebar|>
-      <Sidebar.Header as |Header|>
-        {{#if this.currentCluster.cluster}}
-          <Header.Logo
-            @icon="vault"
-            @route="vault.cluster"
-            @model={{this.currentCluster.cluster.name}}
-            @ariaLabel="home link"
-            data-test-sidebar-logo
-          />
-        {{/if}}
-        {{#if Header.shouldRenderActions}}
-          <Header.Action>
-            <button
-              type="button"
-              class="button is-transparent has-text-white"
-              {{on "click" (fn (mut this.console.isOpen) (not this.console.isOpen))}}
+    <Hds::SideNav @isResponsive={{true}} @hasA11yRefocus={{true}} @a11yRefocusSkipTo="app-main-content">
+      <:header>
+        <Hds::SideNav::Header>
+          <:logo>
+            {{#if this.currentCluster.cluster}}
+              <Hds::SideNav::Header::HomeLink
+                @icon="vault"
+                @route="vault.cluster"
+                @model={{this.currentCluster.cluster.name}}
+                @ariaLabel="home link"
+                data-test-sidebar-logo
+              />
+            {{/if}}
+          </:logo>
+          <:actions>
+            <Hds::SideNav::Header::IconButton
+              @icon="terminal-screen"
+              @ariaLabel="Console toggle"
               data-test-console-toggle
-            >
-              <Icon @name="terminal-screen" @size="24" />
-            </button>
-          </Header.Action>
-          <Header.Action>
+              {{on "click" (fn (mut this.console.isOpen) (not this.console.isOpen))}}
+            />
             <Sidebar::UserMenu />
-          </Header.Action>
-        {{/if}}
-      </Sidebar.Header>
-      {{! this block is where the HcAppFrame::SidenavPanel components render into }}
-      <Sidebar.Content />
-      <Sidebar.Footer>
+          </:actions>
+        </Hds::SideNav::Header>
+      </:header>
+
+      {{! this block is where the Hds::SideNav::Portal components render into }}
+      <:body>
+        <Hds::SideNav::Portal::Target aria-label="sidebar navigation links" />
+      </:body>
+
+      <:footer>
         {{#if (has-feature "Namespaces")}}
           <NamespacePicker @namespace={{@namespace}} />
         {{/if}}
-      </Sidebar.Footer>
-    </HcAppFrame::Sidenav>
+      </:footer>
+    </Hds::SideNav>
   </Frame.Sidebar>
-  <Frame.Main class="is-relative">
+  <Frame.Main id="app-main-content" class="is-relative">
     {{! outlet for app content }}
     <div id="modal-wormhole"></div>
```

Comment:
I don't know how this specific `modal-wormhole` works, but you may want to consider using the modals "container" already provided by the `AppFrame` as target (just pass an ID to it); see how it's done in Cloud UI for example: https://github.com/hashicorp/cloud-ui/blob/b6c5a5a55d5e93143d9606614b4d5a3103bbc610/packages/cloud-ui-core/addon/components/hcp-app-frame/index.hbs#L85

Docs: https://helios.hashicorp.design/layouts/app-frame#modals-container

Reply:
The `pointer-events: none` rule on the `hds-app-frame__modals` class is preventing interactions with the modal content. I could nest the `#modal-wormhole` div in that block and add `pointer-events: all` to it. Not sure if this might be an issue for other consumers and if it would be worth it to remove that rule?

Reply:
> Not sure if this might be an issue for other consumers and if it would be worth it to remove that rule?

Disregard that. Since it is indexed higher up it would swallow all the events.

I'm going to leave as is since we have it on the road map to update to the HDS Modal component at which time this wormhole will be removed.